internal/moduletest/graph/node_state_cleanup.go

```diff
@@ -133,7 +143,7 @@ func (n *NodeStateCleanup) cleanup(ctx *EvalContext, runNode *NodeTestRun, waite
 	// operation. We can just apply the override to the state file and return.
 	if n.applyOverride != nil {
 		runNode.testApply(ctx, variables, waiter)
-		return ctx.GetFileState(n.stateKey).State, nil
+		return ctx.GetFileState(n.stateKey).State, runNode.run.Diagnostics
```

Comment:
I dont think we should return an error here then.

Reply:
That returned error is needed in `func (n *NodeStateCleanup) Execute` to help avoid creating a state artifact when a backend is in use and the cleanup doesn't experience any errors ([see here](https://github.com/hashicorp/terraform/pull/36848/files/e15f40df7a88df3acf777ac6b8fab1b40165fcc3#r2029269219)). I think without that check a state artifact is created whenever something remains in state, and if a backend is in use there will always be something remaining in state.

When I checked requirements (in our 3 person DM with Liam) Liam suggested that a state artifact should be created only if cleanup fails when using a backend.

Reply:
Oh sorry. this is the nested method, not the Execute method.